Mobility GHC-9.2.5 work

arjunkathuria opened the pull request:
`haskell-sequelize` is a dependency of the Mobility project. As a part of the larger upgrade efforts to bring the Mobility set of projects on the newer GHC `9.2.5` platform, this repo needed to be upgraded, in order for it to build with GHC `9.2.5`.

This PR makes changes that enables it to jump from `lts-15.15` with older GHC `8.8.4` to being able to build with newer stackage `lts-20.11` with new GHC `9.2.5`

Apart from being used as a dependency in Mobility set of projects, it is also used in quite a lot of other places, probably because it's a dependency of the fundamental `Euler-hs` framework. As a result, it was imperative that we upgraded this repo and make it build-able with new GHC 9.2.5

**Note**:- Please note that this PR should be merged in a separate `Mobility/GHC-9.2.5` branch which should run as it's own thing for now in parallel. It should **NOT** be merged back in master for now. This version makes use of upstream `Beam` instead of our internal (and outdated) `Beam` forks.

arjunkathuria committed:
Bump stack.yaml to use newest lts-20.4 stackage version

arjunkathuria committed:
Drop old pinned deps version from stack

* removes old pinned package version from stack
* removes the dependency on internal Beam Forks, in favour of upstream Beam

arjunkathuria committed:
Upgrade codebase to Aeson 2 HashMap -> KeyMap syntax

* Updates to new Aeson KeyMap syntax where required to compile

arjunkathuria committed:
remove a qualifying forall

* needed to drop a pair of qualifying foralls for project to compile

arjunkathuria committed:
Bump cabal file version, remove old limits, add changelog

* Bumps the package version in cabal file
* Removes old limits on package versions in cabal file
* Add a new limit for Aeson in cabal file, since we migrated to new syntax
* Adds a Changelog file

arjunkathuria committed:
adds stack yaml lock file

arjunkathuria committed:
Move back to internal beam-mysql fork, newer patched version.

arjunkathuria committed:
update Tests file to remove a unavailable module name

arjunkathuria committed:
commits updated stack.yaml.lock file

arjunkathuria committed:
Update resolver to lts-20.11 from 20.04 in stack.yaml file

* Bumps the stack resolver version from lts-20.4 to lts-20.11

* lts-20.11 because it is the latest stack resolver version
  with GHC-9.2.5 compiler version, newer resolvers than this
  come with GHC 9.2.6 or newer. Keeping things on 9.2.5 for now
  since everything else in Mobility projects is on 9.2.5.

arjunkathuria committed:
chore: update changelog to reflect updated stackage resovler

* Updates the changelog to reflect that we upgraded from
  stackage resolver lts-15.15 -> lts-20.4 -> lts-20.11

aravindgopall commented on stack.yaml:
```diff
@@ -10,26 +12,17 @@ packages:
+  - git: https://github.com/winterland1989/word24.git
+    commit: 445f791e35ddc8098f05879dbcd07c41b115cb39
+
+  - git: https://github.com/arjunkathuria/beam-mysql
```
can this be changed to upstream commit.

arjunkathuria replied:
Ah of-course, good catch.

arjunkathuria committed:
Switch to upstream juspay namespace repository for "beam-mysql"

* Moves away from my own fork of the "beam-mysql" library.
  (now that the changes are merged in "juspay" github org library.)

* Replaces "beam-mysql "library from
  "github.com/arjunkathuria/beam-mysql" to "github.com/juspay/beam-mysql"

* Commits the updated stack.yaml and it's lockfile
